### ISSUE TITLE
Build fails when creating a new target

### ISSUE THREAD

**Issue by TE-HarumiSagawa** [firebase/firebase-ios-sdk#3390]
###  environment

  * Xcode version: _____10.2
  * Firebase Component: _____ (Core, Analytics)

###  problem

Creating a new Build Target causes an error if it does not find FirebaseAnalytics.
Podfile is defined as follows.
```
# Uncomment the next line to define a global platform for your project
platform :ios, '10.0'

target 'ProductName' do
  # Comment the next line if you're not using Swift and don't want to use dynamic frameworks
  use_frameworks!

  # Pods for ProductName
  pod 'libb2s', '1.0.2'
  pod 'RxBluetoothKit','5.2.1'
  pod 'Swinject','2.5.0'
  pod 'iOSDFULibrary','4.2.2'
  pod 'Firebase/Core', '5.20.1'
  pod 'SwiftLint','0.27.0', :configurations => ['Debug']
  pod 'SwiftFormat/CLI','0.35.7', :configurations => ['Debug']
  pod 'CocoaLumberjack/Swift', '3.5.1'
  pod 'SimulatorStatusMagic','2.4.1'
  pod 'Firebase/Analytics', '5.20.1'
  pod 'Fabric', '1.9.0'
  pod 'Crashlytics', '3.12.0'

  # Newly added target
  target 'ProductNameDev' do
    inherit! :search_paths	
  end


  target 'ProductNameTests' do
    inherit! :search_paths
    # Pods for testing
  end

  target 'ProductNameUITests' do
    inherit! :search_paths
  end
end


```

#### error message:

```
Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_FIRAnalytics", referenced from:
      objc-class-ref in FirebaseAnalitics.o
     (maybe you meant: _OBJC_CLASS_$_FIRAnalyticsConfiguration)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)


Showing Recent Messages
:-1: Undefined symbol: _OBJC_CLASS_$_FIRAnalytics
```

**Comment by ryanwilson** [comment]
Hi @TE-HarumiSagawa - version 5.20.1 is about 3 months old now and we've gone on to a new major version (6). Are you able to update to the latest version? If not, you should look at the last release of 5.X which is 5.20.2 (which should have no API changes). Can you reproduce the issue in either of those versions?

**Comment by TE-HarumiSagawa** [comment]
Trying to podinstall 5.20.2 results in an error.
```
[!] CocoaPods could not find compatible versions for pod "Firebase/Core":
  In Podfile:
    Firebase/Analytics (= 5.20.2) was resolved to 5.20.2, which depends on
      Firebase/Core (= 5.20.2)

    Firebase/Core (= 5.20.1)

Specs satisfying the `Firebase/Core (= 5.20.1), Firebase/Core (= 5.20.2)` dependency were found, but they required a higher minimum deployment target.
```

**Comment by paulb777** [comment]
Did you update all of the `Firebase` pod specifications to 5.20.2. It looks like `Firebase/Core` and `Firebase/Analytics` may be different.

If that's not it, you should be able to get more data to diagnose by examining the `Podfile.lock`.
